hash_table/hash_table.go

```diff
@@ -0,0 +1,134 @@
+package hash_table
+
+import (
+	"errors"
+)
+
+type Node struct {
+	key   int
+	value int
+	next  *Node
+}
+
+type HashTable struct {
+	arr  []*Node
+	size int
+}
+
+func NewHashTable() *HashTable {
+	return &HashTable{
+		arr: make([]*Node, 10),
```

Comment:
I prefer to use `8` or `16` as default capacity instead of `10`

Reply:
Is it because it's a power of 2?

Reply:
Yes, we should use a power of 2 so that we can control what capacity will be